Refactor to be more pythonic

sposs opened the pull request.

Commit:
Refactor to make it more pythonic.

Some more changes are possible to have easier build/deploy
(using pbr).

cmweiss commented:
Thanks, I'll take a look at your pull request in detail. It may take me a few days to dig in. Would you please give me a bit more detail on your changes. More pythonic is nice to know but I need more explanation, please.

sposs commented:
I moved the things around (split code and tests), changed the package structure to make it more standard, as there was an unnecessary level. I also fixed some PEP8 conventions. I simplified operations (a = a+1 becomes a += 1). Nothing else.

sposs commented:
#3 is much better than this one...
